src/pages/LandingPages/CorporateCampaign/CCLandingPage.vue

```diff
@@ -711,10 +707,19 @@ export default {
 				// Future usage will not require the promoFundId relying only on the basket id
 				return getPromoFromBasket(targetPromo[0].promoFund?.id, this.apollo);
 			}).then(response => {
-				// TODO Handle response and any potential errors
-				this.promoData = response.data?.shop?.promoCampaign;
-				this.promoApplied = true;
-				this.loadingPromotion = false;
+				// Verify that applied promotion is for current page
+				if (this.verifyPromoMatchesPageId(response.data?.shop?.promoCampaign?.managedAccount?.pageId)) {
+					this.promoData = response.data?.shop?.promoCampaign;
+					this.promoApplied = true;
+					this.loadingPromotion = false;
+					this.promoErrorMessage = null;
+				} else {
+					// Handle response and any potential errors
+					// > this reveals and prior error messages from the promo application
+					this.promoApplied = false;
+					this.loadingPromotion = false;
+				}
```

Comment:
Moved the CampaignStatus update here after we've gone through a basket and promo check.